lib/SILGen/SILGenApply.cpp

```diff
@@ -4239,7 +4247,9 @@ namespace {
         // with specialized emitters above, enum constructors use
         // the AST-level abstraction pattern, to ensure that function
         // types in payloads are re-abstracted correctly.
+#if !defined(_MSC_VER) || defined(__clang__)
         assert(!AssumedPlusZeroSelf);
```

Comment:
We should look into what's making this assertion fail. It could indicate a serious problem.